packages/core/core/src/BundleGraph.js

```diff
@@ -282,6 +299,34 @@ export default class BundleGraph {
               let newNodeId = hashString(
                 node.id + [...target.keys()].join(','),
               );
+
+              let symbols = new Map();
+              for (let [as, from] of target) {
+                let existing = nodeValueSymbols.get(as);
+                if (existing) {
+                  symbols.set(from, existing);
+                } else {
+                  invariant(isReexportAll);
+                  let local = `${node.value.id}$rewrite$${asset}$${from}`;
+                  symbols.set(from, {
+                    isWeak: true,
+                    local,
+                    loc: reexportAllLoc,
+                  });
+                  // It might already exist with multiple export-alls causing ambiguous resolution
+                  if (sourceAssetSymbols && !sourceAssetSymbols.has(as)) {
+                    sourceAssetSymbols.set(as, {
```

Comment:
Why does this change the symbols of some asset in the asset graph, as opposed to the new `symbols` map?

Reply:
So the symbols map is for the current asset, but we need to set the symbols for the source asset using the node id from the current node in the loop. Not sure if there's a better way to do this though

Reply:
I think a big problem is that this actually mutates the asset graph, so running `BundleGraph.fromAssetGraph` multiple times isn't safe (which would happen in watch mode).

So at least it should be something like this: https://github.com/parcel-bundler/parcel/commit/908467af6149a89f27cc925390160dc0ec1bd37e
(Not sure how much the added `new Map(...)` affects performance.)

Reply:
Isn't this still mutating the asset graph? https://github.com/parcel-bundler/parcel/commit/908467af6149a89f27cc925390160dc0ec1bd37e#diff-b2e31de6fe476821294b35d7fb17106cf1fc48b37ebe5800448dcb3608348b6cR306-R315

Reply:
That does (pseudocode):
```js
      let sourceAssetId = translateAssetGraphToBundleGraphNodeId(node.value.sourceAssetId);
      let sourceAsset = bundeGraph.getNode(sourceAssetId)) // a node in the bundle graph
      let sourceAssetSymbols = sourceAsset.value.symbols;
      sourceAssetSymbols.set(...)
```

And then [this](https://github.com/parcel-bundler/parcel/commit/908467af6149a89f27cc925390160dc0ec1bd37e#diff-b2e31de6fe476821294b35d7fb17106cf1fc48b37ebe5800448dcb3608348b6cR367) makes sure that the node is copied before adding to the bundle graph, otherwise the `sourceAssetSymbols.set` would still mutate the asset graph. (As I've said, I hope that copying doesn't affect performance)

Reply:
I see, lgtm then. I didn't see any noticeable perf regressions on our large projects